cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -110,18 +226,50 @@ _kubectl label node $SRIOV_NODE sriov=true
 
 wait_pods_ready
 
-# we need to sleep as the configurations below need to appear
-sleep 30
+# Ensure webook-configuration object created
+wait_k8s_object "validatingwebhookconfiguration" "operator-webhook-config"  || exit 1
+wait_k8s_object "mutatingwebhookconfiguration"   "operator-webhook-config"  || exit 1
+wait_k8s_object "mutatingwebhookconfiguration"   "network-resources-injector-config"  || exit 1
 
 _kubectl patch validatingwebhookconfiguration operator-webhook-config --patch '{"webhooks":[{"name":"operator-webhook.sriovnetwork.openshift.io", "clientConfig": { "caBundle": "'"$(cat $CSRCREATORPATH/operator-webhook.cert)"'" }}]}'
 _kubectl patch mutatingwebhookconfiguration network-resources-injector-config --patch '{"webhooks":[{"name":"network-resources-injector-config.k8s.io", "clientConfig": { "caBundle": "'"$(cat $CSRCREATORPATH/network-resources-injector.cert)"'" }}]}'
 _kubectl patch mutatingwebhookconfiguration operator-webhook-config --patch '{"webhooks":[{"name":"operator-webhook.sriovnetwork.openshift.io", "clientConfig": { "caBundle": "'"$(cat $CSRCREATORPATH/operator-webhook.cert)"'" }}]}'
 
-# we need to sleep to wait for the configuration above the be picked up
-sleep 60
+# Since sriov-operator doesnt have a condition or Status to indicate if
+# 'operator-webhook' and 'network-resources-injector' webhooks  certificates are
+# configured, in order to check if caBundle reconcile is finished it is necessary
+# to wait for the "NoSchedule" taint to present and then absent.
+wait_for_taint "NoSchedule" || true
```

Comment:
can there be a race that lines 242 243 would finish before the cluster managed to create taints and remove them ?
hence we would lose the barrier

Reply:
There will be no race here.
The assumption should be that a taint **might** be created on the node after creating SriovPolicy,
there is no guarantee that there will be a taint, its depends on the environment and the sriov card hardware.
If there is we want it in the logs, if there isnt `wait_for_taint` will finish after the timeout and `wait_for_absence`will return immediately.

Reply:
lets say line 242 + 243 runs on t4, no taint exists on t4
but taint does comes on t5 (t5 > t4),
script is already on other advanced place on t5, the question is would it suffer from it that now there are taints when it checks for other things ?

Reply:
Just to check that we are on the same page,
line 251 (was 242) run until there is a taint or timeout.
then 243 (was 243) run until taint is removed or a timeout
(no simultaneously)
As I said in earlier comment, after patching sriov webhooks, a taint may be created but it is not guaranteed 
thus we wait (about 5 minutes which is enough to verify caBundle picked up)
(next step is crating a policy which may also cause a taint, we look for it in line 277)

Reply:
ye we are sync on that, my question is other, but we can take it offline thanks, you did great enough ;]